sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -112,5 +125,134 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
+                    .AppendLine();
+            }
+
+            if (parseSuccess && data != null && data.Count > 0)
+            {
+                messageBuilder
+                    .AppendLine()
+                    .AppendLine("Additional Information:");
+                foreach (KeyValuePair<string, string> info in data)
+                {
+                    messageBuilder
+                        .Append(info.Key)
+                        .Append(": ")
+                        .AppendLine(info.Value);
+                }
+            }
+
+            if (response.ContentStream is MemoryStream)
+            {
+                string content = response.Content.ToString();
+                if (content.Length > 0)
+                {
+                    messageBuilder
+                        .AppendLine()
+                        .AppendLine("Content:")
+                        .AppendLine(content);
+                }
+            }
+
+            messageBuilder
+                .AppendLine()
+                .AppendLine("Headers:");
+
+            foreach (HttpHeader responseHeader in response.Headers)
+            {
+                string headerValue = response.Sanitizer.SanitizeHeader(responseHeader.Name, responseHeader.Value);
+                messageBuilder.AppendLine($"{responseHeader.Name}: {headerValue}");
+            }
+
+            var formatMessage = messageBuilder.ToString();
+            return (formatMessage, error?.Code, data);
+        }
+
+        /// <summary>
+        /// This is intentionally sync-only as it will only be called by the ctor.
+        /// </summary>
+        /// <param name="response"></param>
+        /// <returns></returns>
+        internal static string? ReadContent(Response response)
+        {
+            string? content = null;
+
+            if (response.ContentStream != null &&
+                ContentTypeUtilities.TryGetTextEncoding(response.Headers.ContentType, out var encoding))
+            {
+                using (var streamReader = new StreamReader(response.ContentStream, encoding))
+                {
+                    content = streamReader.ReadToEnd();
+                }
+            }
+
+            return content;
+        }
+
+        internal static bool TryExtractErrorContent(Response response, out ResponseError? error, out IDictionary<string, string>? data)
+        {
+            error = null;
+            data = null;
+            try
+            {
+                string? content = null;
+                if (response.ContentStream != null && response.ContentStream.CanSeek)
+                {
+                    content = response.Content.ToString();
+                }
+                else
+                {
+                    // this path should only happen in exceptional cases such as when
+                    // the RFE ctor was called directly by client or customer code with an un-buffered response.
+                    // Generated code would never do this.
```

Comment:
What about storage generated code?

Reply:
Once the full set of generator changes come in, this will be true for all error paths

Reply:
Is there an issue filed for updating ResponseBodyPolicy to use IsErrorResponse, and updating generater to use this as well?

Reply:
[yes - Al](https://github.com/Azure/autorest.csharp/issues/2385)